src/App.js

```diff
@@ -1,8 +1,24 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatEntries, setChatEntries] = useState(chatMessages);
+
+  const likeNumber = chatEntries.filter(entry => entry.liked === true).length;
+
+  const recordLikes = (updatedEntry) => {
+    const entries = chatEntries.map((entry) =>
+      {if (updatedEntry.id === entry.id) {
+        return updatedEntry
+      } else {
+        return entry
+      }
+    });
+    setChatEntries(entries)
+  };
```

Comment:
In the same vein while `recordLikes` isn't a bad name, it feels a bit misleading to me, especially in the case where likes are being removed. I'd recommend something like `toggleLiked` that more directly says what happens in this function.

Though, given that the actual toggling happens in `ChatEntry`, it can be argued that `recordLikes` is more accurate. At the same time, it's not as if the `liked` setting actually changes, there's just an updated entry that gets replaced. So that implies to me a name like `updateEntry`.